File: 2021/day6/go/main_test.go
package main

import (
	"testing"
)

var input = "3,4,3,1,2"

func TestPart1(t *testing.T) {
	expected := 5934
	actual := part1(input)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

func TestPart2(t *testing.T) {
	expected := 26984457539
	actual := part2(input)
	if actual != expected {
		t.Errorf("actual [%v] != expected [%v]", actual, expected)
	}
}

File: 2022/day6/js/main.test.js
"use strict";

const assert = require("assert");
const { part1, part2 } = require("./main");

async function test_part1(input) {
  const expected = 7;
  const actual = await part1(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test_part2(input) {
  const expected = 19;
  const actual = await part2(input);
  assert.equal(
    actual,
    expected,
    `actual: [${actual}] != expected [${expected}]`
  );
}

async function test() {
  let input = "mjqjpqmgbljsphdztnvjfqwrcgsmlb";
  await test_part1(input);
  input = "mjqjpqmgbljsphdztnvjfqwrcgsmlb";
  await test_part2(input);
}

test();
